src/main/asciidoc/reference/vault-secret-engines.adoc

```diff
@@ -0,0 +1,252 @@
+<3> Tokens can be renewed through the Token API. Typically, that is done by `SessionManager` to keep track of the Vault session token.
+<5> Tokens can be revoked if needed through the Token API. Typically, that is done by `SessionManager` to keep track of the Vault session token.
+====
+
+You can find more details about the https://www.vaultproject.io/api-docs/auth/token[Vault Token Auth Method API] in the Vault reference documentation.
+
+[[vault.core.backends.transit]]
+== Transit Backend
+
+The transit secrets engine handles cryptographic functions on data in-transit.
+Vault doesn't store the data sent to this secrets engine.
+It can also be seen as "cryptography as a service" or "encryption as a service".
+The transit secrets engine can also sign and verify data; generate hashes and HMACs of data; and act as a random bytes source.
+
+The primary use case for transit is to encrypt data from applications while still storing that encrypted data in some primary data store.
+This relieves the burden of proper encryption/decryption from application developers and pushes the burden onto the operators of Vault.
+
+Spring Vault supports a wide range of Transit operations:
+
+* Key creation
+* Key reconfiguration
+* Encryption/Decryption/Rewrapping
+* HMAC computation
+* Signing and signature verification
+
+All operations within `transit` are centered around keys.
+The Transit engine supports the versioning of keys allowing you to pick a key type https://www.vaultproject.io/docs/secrets/transit[from a variety of key types].
+Note that depending on the key type may impose a limitation on which operations can used.
+
+Let's take a look at encryption and decryption.
+The following examples shows how to create a key and how to encrypt and decrypt data:
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/Transit.java[tags=encryptSimple]
+----
+<1> First, we need a key to begin with.
+Each key requires the type to be specified. `aes128-gcm96` supports encryption, decryption, key derivation, and convergent encryption of which we need encryption and decryption for this example.
+<2> Next, we encrypt a `String` containing the plaintext that should be encrypted.
+The input `String` uses the default `Charset` to encode the string into its binary representation.
+Requesting a token returns a `VaultToken` which is used as value object for Vault tokens.
+The `encrypt` method returns Base64-encoded ciphertext, typically starting with `vault:`.
+<3> To decrypt ciphertext into plaintext, call the `decrypt` method.
+It decrypts the ciphertext and returns a `String` that is decoded using the default charset.
+====
+
+The preceeding example uses simple strings for cryptographic operations.
+While it's a simple approach, it bears the risk of charset misconfiguration and it is not binary-safe.
+Binary safety is required when the plaintext uses a binary representation for data such as images, compressed data or binary datastructures.
```

Comment:
Change to "Binary safety is required when the plain text uses a binary representation for data such as images, compressed data, or binary data structures." (space in "plain text", comma before the last list item, and a space in "data structures").